src/network/networkd-setlink.c

```diff
@@ -680,6 +680,15 @@ int link_request_to_set_mac(Link *link) {
         if (!link->network->mac)
                 return 0;
 
+        if (link->hw_addr.length != sizeof(struct ether_addr)) {
+                log_link_debug(link, "Size of hardware address (%zu) does not match the size of MAC address (%zu), ignoring.",
+                               link->hw_addr.length, sizeof(struct ether_addr));
+                return 0;
+        }
```

Comment:
hmm, so this means the code will only ever set the hw address of ethernet? isn't this a useful feature for other transports, too? infiniband and such, which have longer hw addresses?

Reply:
I do not have and am not familiar with infiniband or longer hw addreses. But, at least, the current logic does not support such devices. Maybe, we should rename `MACAddress=` -> `HardwareAddress=` and support address with an arbitrary length.

Let's do that when people request to support such hw.

Reply:
ok. but maybe add a comment explaining this. i.e. a `/* Note that for now we only support changing hardware addresses on Ethernet. */` or so would probably suffice, next to the check